### ISSUE TITLE
Menu categories - merge all merchants into one & merge all items into one

### ISSUE THREAD

**Issue by FrutyX** [DinoDevs/GladiatusCrazyAddon#430]
![just-do-it-shia-la-beouf](https://user-images.githubusercontent.com/23278552/195979184-9459c53f-9f2b-42a0-9136-7bdc12ec43eb.gif)

**Comment by GramThanos** [comment]
Just an FYI, modifing the page layout may impact the user experience negatively. Consider the following example:

- The page loads and the user instantly scrolls down to click the auction button.
- While his cursor is above the auction button, the addon loads and changes the layout
- The user clicked the right mouse button thinking he will end up at the auction
- Due to the layout change the auction button is now a few pixels up and the user ends up clicking the market button.

This is getting worse the slower/older the PC of the user is. For this reason it is suggested not to alter anything on the page that may affect the possitioning os commonly used items souch as menu links.
If you still want to do so, the changes should be made by the CSS (or atleast a quick CSS change to set a place holder so that the position do not change after the JavaScript loads, even if the content is not there yet). A similar usecase is been implemented when moving the top event banner, where the CSS class to move the layout is been applied before the page is ready.

**Comment by FrutyX** [comment]
I completely understand that, and if it's possible, I always use CSS rather than javascript.

However, keep in mind that these features are optional, if the user does not like it, he can always disable it (it's false by default) and all possible changes will be turned off.

**Comment by GreatApo** [comment]
> However, keep in mind that these features are optional, if the user does not like it, he can always disable it (it's false by default) and all possible changes will be turned off.

I completely agree. I would leave it by default off and let people enable it if they want to.

**Comment by FrutyX** [comment]
I was thinking that maybe I could name those buttons with merchant names (and disable overflow), but I could not find anything in GCA about cached merchant names, they are not used. I only found them in the merchant search, but they are not cached (I think), it seems that their names are loaded during the search request.

I haven't done UI translation caching before, so I am not sure how I should save them...

**Comment by GreatApo** [comment]
> I was thinking that maybe I could name those buttons with merchant names (and disable overflow), but I could not find anything in GCA about cached merchant names, they are not used. I only found them in the merchant search, but they are not cached (I think), it seems that their names are loaded during the search request.
> 

The names are in the menu in every page you visit, you don't really need to cache them

**Comment by FrutyX** [comment]
When they add the dice event menu category, it will mess up the names order... and I haven't been able to lock on the link... just awesome...

**Comment by FrutyX** [comment]
Lmao, nevermind, already found a solution. See? Every time I say there is a problem, i will instantly find a solution within the first five minutes, haha.

**Comment by GreatApo** [comment]
Nice!

It will be more helpful to use icons instead of names in the submenu.
![image](https://user-images.githubusercontent.com/9620510/197387443-c9da37c1-918e-42ba-8741-59c37a3c3f89.png)


We may need to create a menu group in the settings at some point.

**Comment by FrutyX** [comment]
Don't have any icons, unless I would create completely new ones from scratch. I was already searching for some icons in the game UI, but I haven't found anything fitting, and item icons are already used by market and auction menu categories, it would look messy.

For now, this was the best option.

**Comment by FrutyX** [comment]
Of course I could have used emojis like I did with packages shortcuts, but eh, I don't want to overuse emojis too much, plus figuring out a fitting emoji for smelting or magus etc... eh.

**Comment by FrutyX** [comment]
> We may need to create a menu group in the settings at some point.

Done.

**Comment by GreatApo** [comment]
> Done.

Nice rework Fruty!
